test/test_script.py

```diff
@@ -1168,6 +1169,47 @@ def test_cmd_datetime(self):
             areg = r"^Wrong number of arguments for \$datetime: Expected between 0 and 1, "
             with self.assertRaisesRegex(ScriptError, areg):
                 self.parser.eval("$datetime(abc,def)")
+            # The following is disabled because it only raises the exception for Python < 3.7 on Ubuntu
+            # # Tests with invalid format code (ends with % failes on all platforms)
+            # areg = r"^\$datetime:\d+:\d+: Unsupported format code"
+            # with self.assertRaisesRegex(ScriptRuntimeError, areg):
+            #     self.parser.eval(r"$datetime(\%)")
+        except (AssertionError, ValueError, IndexError) as err:
+            raise err
+        finally:
+            # Restore original datetime object
+            datetime.datetime = original_datetime
+
+    def test_cmd_datetime_platform_dependent(self):
+        # Platform dependent testing because different platforms (both os and Python version)
+        # support some format arguments differently.
+        possible_tests = [
+            ('%', r'$datetime(\%)'),        # Hanging % at end of format
+            ('%-d', r'$datetime(\%-d)'),    # Non zero-padded day
+            ('%-m', r'$datetime(\%-m)'),    # Non zero-padded month
+            ('%3Y', r'$datetime(\%3Y)'),    # Length specifier shorter than string
+        ]
+        tests_to_run = []
+        # Get list of tests for unsupported format codes
+        for (test_arg, test_case) in possible_tests:
+            try:
+                datetime.datetime.now().strftime(test_arg)
+            except ValueError:
+                tests_to_run.append(test_case)
```

Comment:
`test_case` can be built from `test_arg`, so I think we can simplify `possible_tests` (and make it immutable while at it):

```python
possible_tests = (
  '%',  # Hanging % at end of format
...

tests_to_run.append(r"$datetime(\%s)" % test_arg)
```

Reply:
I'll make the change.  I thought about that originally, but thought it might be an issue if the argument didn't need to start with an escape character.  As it turns out, everything we're (currently) testing is escaped so it's not a problem.